File: 03-poo/04-encapsulamiento-y-modificadores-acceso/scala/ejemplos/Encapsulamiento.scala
class Cuenta:
  private var saldo = 0.0
  def depositar(monto: Double): Unit = if monto > 0 then saldo += monto
  def saldoActual: Double = saldo

@main def encapsulamiento(): Unit =
  val cuenta = new Cuenta
  cuenta.depositar(100)
  println(cuenta.saldoActual)

File: 03-poo/04-encapsulamiento-y-modificadores-acceso/java/ejemplos/Encapsulamiento.java
public class Encapsulamiento {
    static class Cuenta {
        private double saldo;

        void depositar(double monto) {
            // La clase controla cómo cambia su propio estado.
            if (monto > 0) saldo += monto;
        }

        double getSaldo() { return saldo; }
    }
    public static void main(String[] args) {
        Cuenta cuenta = new Cuenta();
        cuenta.depositar(100);
        System.out.println(cuenta.getSaldo());
    }
}
